src/components/WorkOrder/AppointmentDetails.js

```diff
@@ -41,36 +42,24 @@ const AppointmentDetails = ({ workOrder, schedulerSessionId }) => {
     if (workOrder.externalAppointmentManagementUrl) {
       if (schedulerSessionId) {
         return (
-          <Link
-            href={`${workOrder.externalAppointmentManagementUrl}&sessionId=${schedulerSessionId}`}
-          >
-            <a
-              className="lbh-link"
-              target="_blank"
-              rel="noopener"
-              onClick={openExternalLinkEventHandler}
-            >
-              <strong>Open DRS</strong> to{' '}
-              {hasExistingAppointment ? 'reschedule' : 'book an'} appointment
-            </a>
-          </Link>
+          <ScheduleDRSAppointmentLink
+            workOrder={workOrder}
+            schedulerSessionId={schedulerSessionId}
+            openLinkEventHandler={openExternalLinkEventHandler}
+            hasExistingAppointment={hasExistingAppointment}
+            appointmentIsToday={workOrder.appointmentIsToday()}
+          />
         )
       } else {
         console.error('Scheduler Session ID does not exist')
       }
     } else {
-      const href = hasExistingAppointment
-        ? `/work-orders/${workOrder.reference}/appointment/edit`
-        : `/work-orders/${workOrder.reference}/appointment/new`
-
-      const linkText = hasExistingAppointment
-        ? 'Reschedule appointment'
-        : 'Schedule appointment'
-
       return (
-        <Link href={href}>
-          <a className="lbh-link">{linkText}</a>
-        </Link>
+        <ScheduleInternalAppointmentLink
+          workOrderReference={workOrder.reference}
+          hasExistingAppointment={hasExistingAppointment}
+          appointmentIsToday={workOrder.appointmentIsToday()}
+        />
       )
```

Comment:
That's a great improvement moving them into separate components. I am wondering if it's worth to add another component `ScheduleAppointmentLink` which will have the if logic, so we have only one component here.

Reply:
Thanks for this - I think this is a great shout but are you OK if I implement this in a separate PR another time? 

I'm keen to get this story over to QA as it has been delayed for a few different reasons.

Reply:
Agree with Evangelos, and it is fine by me if we implement it in a separate PR

Reply:
Yeah sure, we can pick up this on a separate PR.